src/resources/resources.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from "@nestjs/common";
+
+import { EntityManager } from "@mikro-orm/postgresql";
+
+import { ClassroomsRepository } from "@/classrooms/classrooms.repository";
+import { FileUploadsService } from "@/file-uploads/file-uploads.service";
+
+import { AddResourcesDto } from "./resources.dtos";
+import { ResourcesRepository } from "./resources.repository";
+
+@Injectable()
+export class ResourcesService {
+  constructor(
+    private readonly resourcesRepository: ResourcesRepository,
+    private readonly fileUploadsService: FileUploadsService,
+    private readonly classroomsRepository: ClassroomsRepository,
+    private readonly em: EntityManager,
+  ) {}
+
+  async addResources(
+    classroomId: number,
+    addResourcesDto: AddResourcesDto,
+    file: Express.Multer.File,
+  ) {
+    const classroom = await this.classroomsRepository.findOneOrFail({ id: classroomId });
+
+    const fileUrl = await this.fileUploadsService.handleFileUpload(file);
+
+    const resource = this.resourcesRepository.create({
+      ...addResourcesDto,
+      fileUrl,
+      classroom: classroom.id,
+    });
+
+    await this.em.persistAndFlush(resource);
```

Comment:
```suggestion
    await this.resourcesRepository.getEntityManager().persistAndFlush(resource);
```

Best to use em scoped to the entity repository itself rather than the global one. That way transactions only persist for that specific context and not for the global context.